uobtheatre/payments/payables.py

```diff
@@ -151,59 +157,96 @@ def refund(self, authorizing_user: User, do_async=True, send_admin_email=True):
                 html_message=mail.to_html(),
             )
 
-    def complete(self, payment: Transaction = None):
+    @property
+    @abc.abstractmethod
+    def total(self):
+        """The total amount required to pay for the payable"""
+        raise NotImplementedError
+
+    @property
+    @abc.abstractmethod
+    def misc_costs_value(self):
+        """The total platform fee, in pence, for this payable"""
+        raise NotImplementedError
+
+    def pay(self, payment_method: "PaymentProvider") -> Optional["Transaction"]:
         """
-        Called once the pay object has been completly paid for. Payment passed is the finishing transaction
+        Pay for booking using provided payment method.
+
+        Args:
+            payment_method (PaymentMethod): The payment method used to pay for
+                the booking
+
+        Returns:
+            Payment: The payment created by the checkout (optional)
+
+        Raises:
+            CantBePaidForException: If the status of the booking is not
+                IN_PROGRESS
         """
+        if self.status != Payable.Status.IN_PROGRESS:
+            raise CantBePaidForException(
+                message=f"A payable with status {self.get_status_display()} cannot be paid for"
+            )
+
+        # Cancel and delete pending payments for this booking
+        for payment in self.transactions.filter(status=Transaction.Status.PENDING):
+            payment.cancel()
+
+        payment = payment_method.pay(self.total, self.misc_costs_value, self)
+
+        # If a payment is created set the booking as paid
+        if payment.status == Transaction.Status.COMPLETED:
+            self.complete(payment)
+
+        return payment
+
+    def complete(self, payment: Transaction = None):  # pylint: disable=unused-argument
+        """
+        Called once the pay object has been completly paid for. Payment passed
+        is the finishing transaction
+        """
+        self.status = Payable.Status.PAID
+        self.save()
 
     @property
-    def total_sales(self) -> int:
-        """The amount paid by the user for this object."""
-        return self.transactions.annotate_sales_breakdown(["total_sales"])[  # type: ignore
-            "total_sales"
-        ]
+    def total_payments(self) -> int:
+        """The amount paid by the user for this object. (This does not include refunds)"""
```

Comment:
More descriptive even if it is ovbious
```suggestion
        """
        The positive amounts paid by the user for this object. 
         
        - This does not include refunds.
        - This does include the square fee.
        """
```